### DISCUSSION TITLE
PACS Server with Fo-dicom?

### DISCUSSION THREAD

**Discussion by barna21** [fo-dicom/fo-dicom#1274] [Q&A]
Hi,
Any idea how to use FO DICOM to develop a pacs server?

Thanks

**Comment by barna21** [comment]
any idea please?

**Reply by mrbean-bremen** [reply]
You can use fo-dicom to implement the services needed for a PACS, such as the query/retrieve SCP - simple examples can be found in the [sample repository](https://github.com/fo-dicom/fo-dicom-samples/tree/master/Desktop). I don't know what you need exactly - I doubt that you need to implement a complete PACS server, that would be a bit much, and there are enough PACS servers out there. Can you elaborate a bit on what you actually need?

**Reply by barna21** [reply]
Thanks for you relply,
I  need to be able to send and retrieve DICOM images to that PACS app, without any other features.
Thanks.

**Reply by mrbean-bremen** [reply]
Ok, so you need to implement a query/retrieve SCP - check out [this example](https://github.com/fo-dicom/fo-dicom-samples/tree/master/Desktop/QueryRetrieve%20SCP).

**Reply by barna21** [reply]
Thank you so much for your response!

I apologize, I was probably not clear enough. As far as I understand, your suggestions refers more to a DICOM client rather than to a server.
What I would actually need is to be able to develop an app so it can store the DICOM images for several days (it should become a buffer type DICOM storage).
Therefore, I need to send the images from a client app to this "PACS buffer server" and then to have the possibility of accessing it for few days in order to download the images from there. 

I hope I made myself clear this time, but please let me know if you have other queries.

**Reply by mrbean-bremen** [reply]
Actually my suggestion refers to a DICOM server (I was pointing to the examples for the SCP, not the SCU), but you also need a Storage SCP server, of course - there is also [an example](https://github.com/fo-dicom/fo-dicom-samples/tree/master/Desktop/C-Store%20SCP) for that. What you do with the DICOM files, e.g. how you save them, if you add them to a database (certainly makes sense for PACS) etc. is up to you. Regarding fo-dicom, you mainly have to configure the services, and implement the respective callbacks, as shown in the (simple) examples.

**Reply by barna21** [reply]
Ok. I will have a look at your suggestions. Many thanks for your reply.
